core/frontend/src/render/webgl/glsl/AmbientOcclusion.ts

```diff
@@ -152,21 +165,73 @@ vec3 computeNormalFromDepth(vec3 viewPos, vec2 tc, vec2 pixelSize) {
 }
 `;
 
+const computeNonLinearDepthPB = `
+float computeNonLinearDepth(float linearDepth) {
+  return mix(u_frustum.y, u_frustum.x, linearDepth);
+}
+`;
+const computeNonLinearDepthDB = `
+float computeNonLinearDepth(float depth) {
+  return 0.0 == u_logZ.x ? depth * u_logZ.y : exp(depth * u_logZ.y) / u_logZ.x;
+}
+`;
+
+const readDepthPB = `
+float readDepth(vec2 tc) {
+  return readDepthAndOrder(tc).y;
+}
+`;
+const readDepthDB = `
+float readDepth(vec2 tc) {
+  return TEXTURE(u_depthBuffer, tc).r;
+}
+`;
+const unfinalizeLinearDepthDB = `
+  float unfinalizeLinearDepth(float depth) {
+    float eyeZ = 0.0 == u_logZ.x ? depth * u_logZ.y : exp(depth * u_logZ.y) / u_logZ.x;
+    float near = u_frustum.x, far = u_frustum.y;
+    float depthRange = far - near;
+    float linearDepth = (eyeZ - near) / depthRange;
+    return 1.0 - linearDepth;
+  }
+`;
+
+function _shouldUseDB() { return System.instance.supportsLogZBuffer && System.instance.capabilities.supportsTextureFloat; }
```

Comment:
So, not iOS?

Reply:
I did not manage to build + deploy display-test-app with this branch, but the new approach should work on iOS 15+ because that iOS supports WebGL2.  If WebGL1, my latest gen iPad Pro supports frag_depth extension too.

Reply:
> my latest gen iPad Pro supports frag_depth extension too

But only half float, right?
Just curious - if folks want better AO on iOS, they can upgrade to 15.

Reply:
1st gen iPad Pro supports frag_depth under WebGL1 as well.

Reply:
frag_depth is not the issue.

Reply:
Half float, yeah.  Actually half float everywhere on iOS for iTwin.js because of this workaround: https://github.com/iTwin/itwinjs-core/blob/f000d2e163d192261f786081bbbedb16c11bac21/core/webgl-compatibility/src/Capabilities.ts#L276

Reply:
> frag_depth is not the issue.

Elaborate?

Reply:
> Elaborate?

I am not asking about iOS because of frag_depth. I am asking because of OES_texture_float.

Reply:
> > Elaborate?
> 
> I am not asking about iOS because of frag_depth. I am asking because of OES_texture_float.

Okay, thanks.  Yeah.  The OES_texture_float extension means "We can bind float textures as readable."  The half float textures limitation on iOS is for rendering-to-float textures.  Binding existing full float textures (depth in this case - actually 24+8 stencil, I think) as readable is fine.  In retrospect, that isn't even really full float...  I can try to test this on iOS, though.... hopefully I can build display-test-app....

Reply:
Resolve if this looks good?